scripts/wasmtestreport.py

```diff
@@ -154,6 +154,129 @@ def add_test_result(result, file_path, status, error_type, output):
             result["Unknown_Failure"].append(file_path)
 
 
+# ----------------------------------------------------------------------
+# Class: TestResultHandler
+#
+# Purpose:
+#   Centralized handler for test result patterns
+#
+# Variables:
+# - Input: 
+#   result_dict (dict) : the results dictionary.
+#   source_file (str): the test file path
+
+# - Output: Provides methods to add different types of test results; returns True/False
+# ----------------------------------------------------------------------
+class TestResultHandler:
+    """Centralized handler for test result patterns"""
+    
+    def __init__(self, result_dict, source_file):
+        self.result = result_dict
+        self.source_file = str(source_file)
+    
+    def add_success(self, output=""):
+        add_test_result(self.result, self.source_file, "Success", None, output)
+    
+    def add_compile_failure(self, error_msg, is_native=False):
+        error_type = "Failure_native_compiling" if is_native else "Lind_wasm_compiling"
+        add_test_result(self.result, self.source_file, "Failure", error_type, error_msg)
+    
+    def add_runtime_failure(self, error_msg, is_native=False):
+        error_type = "Failure_native_running" if is_native else "Lind_wasm_runtime"
+        add_test_result(self.result, self.source_file, "Failure", error_type, error_msg)
+    
+    def add_timeout(self, output, is_native=False):
+        error_type = "Native_Timeout" if is_native else "Lind_wasm_Timeout"
+        add_test_result(self.result, self.source_file, "Failure", error_type, output)
+    
+    def add_segfault(self, output, is_native=False):
+        error_type = "Native_Segmentation_Fault" if is_native else "Lind_wasm_Segmentation_Fault"
+        add_test_result(self.result, self.source_file, "Failure", error_type, output)
+    
+    def handle_return_code(self, returncode, output, is_native=False):
+        """Handle return code patterns consistently"""
+        if returncode == 0:
+            return True  # Success case, let caller handle
+        elif returncode == "timeout":
+            self.add_timeout(output, is_native)
+        elif returncode == "unknown_error":
+            self.add_runtime_failure(output, is_native)
+        elif is_segmentation_fault(returncode):
+            self.add_segfault(output, is_native)
+        else:
+            add_test_result(self.result, self.source_file, "Failure", "Unknown_Failure", output)
+        return False
+
+
+# ----------------------------------------------------------------------
+# Function: compile_and_run_native
+#
+# Purpose:
+#   Compile and run native version of a test
+#
+# Variables:
+# - Input: source_file - path to the .c file, timeout_sec - timeout for execution
+# - Output: (success, output, returncode, error_type) tuple
+# ----------------------------------------------------------------------
+def compile_and_run_native(source_file, timeout_sec=DEFAULT_TIMEOUT):
```

Comment:
There are two spots here where `run_subprocess` is called with file paths.  If they're always absolute, no issues, but if a relative path sneaks through somehow, running with `cwd=LIND_FS_ROOT` could lead to some confusing issues where it can't find the file.

Might be worth adding an early check to see if the path is absolute or not and failing if it isn't?